source/opt/inline_pass.cpp

```diff
@@ -270,30 +319,120 @@ std::unique_ptr<BasicBlock> InlinePass::AddGuardBlock(
   return new_blk_ptr;
 }
 
+DebugScope InlinePass::BuildDebugScope(const DebugScope& callee_instr_scope,
+                                       DebugInlinedAtContext& inlined_at_ctx) {
+  return DebugScope(callee_instr_scope.GetLexicalScope(),
+                    BuildDebugInlinedAtChain(callee_instr_scope.GetInlinedAt(),
+                                             inlined_at_ctx));
+}
+
+uint32_t InlinePass::BuildDebugInlinedAtChain(
+    uint32_t callee_inlined_at, DebugInlinedAtContext& inlined_at_ctx) {
+  if (inlined_at_ctx.FunctionCallScope().GetLexicalScope() == kNoDebugScope)
+    return kNoInlinedAt;
+
+  // Reuse the already generated DebugInlinedAt chain if exists.
+  uint32_t already_generated_chain_head_id =
+      inlined_at_ctx.InlinedAtChainHead(callee_inlined_at);
+  if (already_generated_chain_head_id != kNoInlinedAt) {
+    return already_generated_chain_head_id;
+  }
+
+  const uint32_t inlined_to_caller =
```

Comment:
This is another case where the variable name is not very useful.  Something like `new_dbg_inlined_at_id` might be better.  Again `inlined_to_caller` reads like it should be a boolean value.